src/overlays/actors/ovl_En_Dekubaba/z_en_dekubaba.c

```diff
@@ -121,85 +133,1191 @@ static ColliderJntSphInit D_809E8FC8 = {
+
+    this->collider.base.colType = COLTYPE_HARD;
+    this->collider.base.acFlags |= AC_HARD;
+    this->timer = 45;
+
+    for (i = 1; i < 7; i++) {
+        element = &this->collider.elements[i];
+        element->dim.worldSphere.center.x = this->actor.world.pos.x;
+        element->dim.worldSphere.center.y = (s16)this->actor.world.pos.y - 7;
+        element->dim.worldSphere.center.z = this->actor.world.pos.z;
+    }
+
+    this->actionFunc = EnDekubaba_Wait;
+}
+
+void EnDekubaba_SetupGrow(EnDekubaba* this) {
+    s32 i;
+
+    Animation_Change(&this->skelAnime, &gDekuBabaFastChompAnim,
+                     Animation_GetLastFrame(&gDekuBabaFastChompAnim) * (1.0f / 15), 0.0f,
+                     Animation_GetLastFrame(&gDekuBabaFastChompAnim), 2, 0.0f);
+
+    this->timer = 15;
+
+    for (i = 2; i < 7; i++) {
+        this->collider.elements[i].info.ocElemFlags |= OCELEM_ON;
+    }
+
+    this->collider.base.colType = COLTYPE_HIT6;
+    this->collider.base.acFlags &= ~AC_HARD;
+    Audio_PlayActorSound2(&this->actor, NA_SE_EN_DUMMY482);
+    this->actionFunc = EnDekubaba_Grow;
+}
+
+void EnDekubaba_SetupRetract(EnDekubaba* this) {
+    s32 i;
+
+    Animation_Change(&this->skelAnime, &gDekuBabaFastChompAnim, -1.5f, Animation_GetLastFrame(&gDekuBabaFastChompAnim),
+                     0.0f, 2, -3.0f);
+
+    this->timer = 15;
+
+    for (i = 2; i < 7; i++) {
+        this->collider.elements[i].info.ocElemFlags &= ~OCELEM_ON;
+    }
+
+    this->actionFunc = EnDekubaba_Retract;
+}
+
+void EnDekubaba_SetupDecideLunge(EnDekubaba* this) {
+    this->timer = Animation_GetLastFrame(&gDekuBabaFastChompAnim) * 2;
+    Animation_MorphToLoop(&this->skelAnime, &gDekuBabaFastChompAnim, -3.0f);
+    this->actionFunc = EnDekubaba_DecideLunge;
+}
+
+void EnDekubaba_SetupPrepareLunge(EnDekubaba* this) {
+    this->timer = 8;
+    this->skelAnime.playSpeed = 0.0f;
+    this->actionFunc = EnDekubaba_PrepareLunge;
+}
+
+void EnDekubaba_SetupLunge(EnDekubaba* this) {
+    Animation_PlayOnce(&this->skelAnime, &gDekuBabaPauseChompAnim);
+    this->timer = 0;
+    this->actionFunc = EnDekubaba_Lunge;
+}
+
+void EnDekubaba_SetupPullBack(EnDekubaba* this) {
+    Animation_Change(&this->skelAnime, &gDekuBabaPauseChompAnim, 1.0f, 15.0f,
+                     Animation_GetLastFrame(&gDekuBabaPauseChompAnim), 2, -3.0f);
+    this->timer = 0;
+    this->actionFunc = EnDekubaba_PullBack;
+}
+
+void EnDekubaba_SetupRecover(EnDekubaba* this) {
+    this->timer = 9;
+    this->collider.base.acFlags |= AC_ON;
+    this->skelAnime.playSpeed = -1.0f;
+    this->actionFunc = EnDekubaba_Recover;
+}
+
+void EnDekubaba_SetupHit(EnDekubaba* this, s32 arg1) {
+    Animation_MorphToPlayOnce(&this->skelAnime, &gDekuBabaPauseChompAnim, -5.0f);
+    this->timer = arg1;
+    this->collider.base.acFlags &= ~AC_ON;
+    Actor_SetScale(&this->actor, this->size * 0.01f);
+
+    if (arg1 == 2) {
+        Actor_SetColorFilter(&this->actor, 0, 155, 0, 62);
```

Comment:
`Actor_SetColorFilter` doesn't take in rgb unfortunately, currently in the code base all its params stay as hex

also on two other lines

Reply:
I *think* @fig02 told me once the args should be hex, dec, hex, dec, although I may be misremembering.

Reply:
yea, these are the args the function takes:
`void Actor_SetColorFilter(Actor* actor, s16 colorFlag, s16 colorIntensityMax, s16 xluFlag, s16 duration);`
color intensity and duration makes sense in decimal. 
Color flag is going to need a macro to pack the bits eventually, so it doesnt matter what it is. hex is what they all are now. And xlu will either be a macro or just true and false. we havent figured that out yet.

Reply:
So, the way it is now is fine since both flags are just 0. no need to make it 0x0